src/test_typing_extensions.py

```diff
@@ -2737,6 +2737,28 @@ def close(self):
         self.assertIsSubclass(B, Custom)
         self.assertNotIsSubclass(A, Custom)
 
+    @skipUnless(
+        hasattr(collections.abc, "Buffer"),
+        "needs collections.abc.Buffer to exist"
+    )
+    @skip_if_py312b1
```

Comment:
This test isn't run in CI at all right now, but I've confirmed locally that it passes with the CPython `main` branch (so when 3.12b2 comes along, it should pass in CI for us here)